packages/core/src/odata/v2/request-builder/update-request-builder.ts

```diff
@@ -42,11 +42,22 @@ export class UpdateRequestBuilderV2<EntityT extends EntityV2>
     this.requestConfig.eTag = _entity.versionIdentifier;
     this.required = new Set<string>();
     this.ignored = new Set<string>();
+
+    this.requestConfig.keys = oDataUriV2.getEntityKeys(
+      this._entity,
+      this._entityConstructor
+    );
+
+    this.requestConfig.payload = this.getPayload();
+  }
+
+  get entity(): EntityT {
+    return this._entity;
   }
 
   /**
+   * @deprecated Since v1.29.0. This method should never be called, it has severe side effects.
```

Comment:
The side effect is that it sets the payload and keys? I totally agree that methods taking no arguments and returning no arguments are rubbish, but perhaps be a bit milder on the deprication. 

Question: Could this be a breaking change. I do not have the complete overview, but in the old implementation the keys were set directly before the request was exectuted. So I could create a update request builder and change the keys of the entities inside and then the execute would take the changed key. This is now not possible anymore, since we have it in the constructor. I think this is better and intended that way but if somebody relies on changing the keys?

Reply:
As discussed on the PR, leaving is as is, because this cannot happen if our API is used.